tests/components/websocket_api/test_messages.py

```diff
@@ -33,18 +35,49 @@ def _event_listener(event):
 
     assert msg0 != msg1
 
-    cache_info = cached_event_message.cache_info()
+    cache_info = lru_event_cache.cache_info()
     assert cache_info.hits == 2
     assert cache_info.misses == 2
     assert cache_info.currsize == 2
 
     cached_event_message(2, events[1])
-    cache_info = cached_event_message.cache_info()
+    cache_info = lru_event_cache.cache_info()
     assert cache_info.hits == 3
     assert cache_info.misses == 2
     assert cache_info.currsize == 2
 
 
+async def test_cached_event_message_with_different_idens(hass):
+    """Test that we cache event messages when the subscrition idens differ."""
+
+    events = []
+
+    @callback
+    def _event_listener(event):
+        events.append(event)
+
+    hass.bus.async_listen(EVENT_STATE_CHANGED, _event_listener)
+
+    hass.states.async_set("light.window", "on")
+    await hass.async_block_till_done()
+
+    assert len(events) == 1
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6372789)